src/pages/profiles/forms/ProfileRename.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { FC, useState } from "react";
+import { Button, Input } from "@canonical/react-components";
+import { useNotify } from "context/notify";
+import { useNavigate } from "react-router-dom";
+import { LxdProfile } from "types/profile";
+import { renameProfile } from "api/profiles";
+import SubmitButton from "components/SubmitButton";
+
+interface Props {
+  profile: LxdProfile;
+  project: string;
+  closeForm: () => void;
+}
+
+const ProfileRename: FC<Props> = ({ profile, project, closeForm }) => {
+  const navigate = useNavigate();
+  const notify = useNotify();
+  const [name, setName] = useState(profile.name);
+  const [isSubmitting, setSubmitting] = useState(false);
+
+  const handleSubmit = () => {
+    if (profile.name === name) {
+      closeForm();
+      return;
+    }
+    setSubmitting(true);
+    renameProfile(profile.name, name, project)
+      .then(() => {
+        navigate(
+          `/ui/${project}/profiles/detail/${name}`,
+          notify.queue(notify.success("Profile renamed."))
+        );
+        closeForm();
+      })
+      .catch((e) => notify.failure("", e, undefined, "Renaming failed"))
```

Comment:
This should be updated according to the new order of parameters for the `failure` function, as set in #272.